### PR TITLE
[MAINTENANCE] Pandas and SqlAlchemy 2.0 follow-up 

### PR DESCRIPTION
**Pull request by Shinnnyshinshin**
* Follow up to #8101 - one `docs-integration` test that needed to be updated.
* `docs-integration` pipeline will use constraints and more specific installation requirements. In the case of non-supported backends (like `redshift` and `bigquery`) we use constraints files and ensure that `pandas < 2.0.0` and `sqlalchemy < 2.0.0` are installed
* Clean up of `docs-integration` tests : requirements like Spark are better organized, and `pyarrow` requirements is applied to more general situations (we want to bring it in even if we aren't using `dremio`)
* Removal of V2 to V3 migration guide tests and fixtures, which are no longer supported

### CODE REVIEW

File: tests/integration/test_definitions/spark/integration_tests.py [hunk 1]
```diff
@@ -21,6 +21,13 @@
         user_flow_script="tests/integration/docusaurus/connecting_to_your_data/in_memory/spark_python_example.py",
         backend_dependencies=[BackendDependencies.SPARK],
     ),
+    IntegrationTestFixture(
+        name="how_to_configure_a_spark_datasource",
+        user_flow_script="tests/integration/docusaurus/connecting_to_your_data/datasource_configuration/how_to_configure_a_spark_datasource.py",
+        data_context_dir="tests/integration/fixtures/no_datasources/great_expectations",
+        data_dir="tests/test_sets/taxi_yellow_tripdata_samples/samples_2020",
+        backend_dependencies=[BackendDependencies.SPARK],
```
Comment (Shinnnyshinshin) [inline tests/integration/test_definitions/spark/integration_tests.py:29]: For some reason this was never running with the `Spark` tests. Moved to appropriate place.

File: reqs/requirements-dev-contrib.txt [hunk 2]
```diff
@@ -4,3 +4,4 @@ invoke>=2.0.0
 mypy==1.3.0
 pre-commit>=2.21.0
 ruff==0.0.271
+pyarrow
```
Comment (Shinnnyshinshin) [inline reqs/requirements-dev-contrib.txt]: `pyarrow` is required for `pandas` Parquet tests, which are non-backend specific.

File: reqs/requirements-dev-dremio.txt [hunk 3]
```diff
@@ -1,3 +1,2 @@
-pyarrow
```
Comment (Shinnnyshinshin) [inline reqs/requirements-dev-dremio.txt:1]: Moved from here to `contrib`

File: tests/integration/docusaurus/reference/core_concepts/result_format.py [hunk 4]
```diff
@@ -224,8 +224,9 @@
         {"my_var": "D", "pk_column": "six"},
         {"my_var": "D", "pk_column": "seven"},
     ],
-    "unexpected_index_query": [3, 4, 5, 6, 7],
+    "unexpected_index_query": "df.filter(items=[3, 4, 5, 6, 7], axis=0)",
```
Comment (Shinnnyshinshin) [inline tests/integration/docusaurus/reference/core_concepts/result_format.py:227]: This is a follow up to #8101

File: ci/azure-pipelines-docs-integration.yml [hunk 5]
```diff
@@ -251,15 +254,16 @@ stages:
...
           displayName: 'Install dependencies'
 
         - script: |
             sqlcmd -U sa -P "ReallyStrongPwd1234%^&*" -Q "CREATE DATABASE test_ci;" -o create_db_output.txt
           displayName: 'Create MSSQL database test_ci'
 
         - script: |
-            pytest -v --docs-tests -m integration --mssql tests/integration/test_script_runner.py
+            # Note expect_column_values_to_only_contain_vowels is skipped because metrics are not implemented for MSSQL.
+            pytest -v --docs-tests -m integration --mssql -k "not test_docs[expect_column_values_to_only_contain_vowels]" tests/integration/test_script_runner.py
```
Comment (Shinnnyshinshin) [inline ci/azure-pipelines-docs-integration.yml:262]: ❤️

File: docs/sphinx_api_docs_source/public_api_missing_threshold.py [hunk 6]
```diff
@@ -33,18 +33,14 @@
     "File: great_expectations/data_context/store/tuple_store_backend.py Name: TupleS3StoreBackend",
     "File: great_expectations/data_context/store/validations_store.py Name: ValidationsStore",
     "File: great_expectations/data_context/types/base.py Name: update",
-    "File: great_expectations/dataset/pandas_dataset.py Name: PandasDataset",
```
Comment (Shinnnyshinshin) [inline docs/sphinx_api_docs_source/public_api_missing_threshold.py:36]: Got to get rid of these after removing the v2_v3 migration code.

File: pyproject.toml [hunk 7]
```diff
@@ -518,24 +518,6 @@ filterwarnings = [
     # pytest.PytestUnraisableExceptionWarning: Exception ignored in: <socket.socket fd=-1, family=AddressFamily.AF_INET, type=SocketKind.SOCK_STREAM, proto=6>
     "ignore: Exception ignored in:UserWarning",
 
-    # SQLAlchemy
```
Comment (Shinnnyshinshin) [inline pyproject.toml:521]: 🚀

File: tests/expectations/core/test_expect_queried_column_value_frequency_to_meet_threshold.py [hunk 8]
```diff
@@ -177,7 +177,7 @@ def test_expect_queried_column_value_frequency_to_meet_threshold_override_query_
     [
         (True, 0.6481340441736482, None, False),
         (False, 0.4791666666666667, 'col("Age")<18', True),
-        (True, 0.6614626129827444, 'col("Age")>17', True),
+        (True, 0.6574585635359116, 'col("Age")>17', True),
```
Comment (Shinnnyshinshin) [inline tests/expectations/core/test_expect_queried_column_value_frequency_to_meet_threshold.py]: Adjusted after the location of the fixture to the one in `test_sets`

File: tests/test_utils.py [hunk 9]
```diff
@@ -686,15 +686,16 @@ def load_data_into_test_database(  # noqa: PLR0912
...
-            add_dataframe_to_db(
-                df=all_dfs_concatenated,
-                name=table_name,
-                con=engine,
-                schema=schema_name,
-                index=False,
-                if_exists="append",
-                method=to_sql_method,
-            )
+            with engine.connect() as connection:
```
Comment (Shinnnyshinshin) [inline tests/test_utils.py:689]: This change brings the `test_utils` file in line with the pattern we are using elsewhere, ie using the context manager

File: tests/profile/test_user_configurable_profiler_v3_batch_request.py [hunk 10]
```diff
@@ -980,12 +980,6 @@ def test_profiler_all_expectation_types_pandas(
     assert results["success"]
 
 
-@pytest.mark.skipif(
```
Comment (Shinnnyshinshin) [inline tests/profile/test_user_configurable_profiler_v3_batch_request.py:983]: This was an unexpected win, but we'll take it:)

File: tests/integration/docusaurus/deployment_patterns/databricks_deployment_patterns_dataframe_yaml_configs.py [hunk 11]
```diff
@@ -142,7 +143,7 @@
 validator.expect_column_values_to_not_be_null(column="passenger_count")
 
 validator.expect_column_values_to_be_between(
-    column="congestion_surcharge", min_value=0, max_value=1000
```
Comment (Shinnnyshinshin) [inline tests/integration/docusaurus/deployment_patterns/databricks_deployment_patterns_dataframe_yaml_configs.py:145]: updated to use `Spark` DF directly. And adjusted the column to a better one (`congestion_surcharge` is always 0)

File: reqs/requirements-dev-contrib.txt (hunk 2 above)
Comment (Shinnnyshinshin) [reply reqs/requirements-dev-contrib.txt]: `FAILED tests/integration/test_script_runner.py::test_docs[how_to_connect_to_in_memory_data_using_pandas]` This is the result

Comment (Shinnnyshinshin) [reply reqs/requirements-dev-contrib.txt]: https://docs.greatexpectations.io/docs/guides/connecting_to_your_data/fluent/in_memory/how_to_connect_to_in_memory_data_using_pandas

File: ci/azure-pipelines-docs-integration.yml [hunk 12]
```diff
@@ -423,7 +431,11 @@ stages:
           displayName: 'Use Python $(python.version)'
 
         - script: |
-            pip install --constraint constraints-dev.txt ".[dev]" pytest-azurepipelines
+            cp constraints-dev.txt constraints-dev-temp.txt
+            echo "SQLAlchemy<2.0.0" >> constraints-dev-temp.txt
```
Comment (Shinnnyshinshin) [inline ci/azure-pipelines-docs-integration.yml:435]: Add better description of constraints-dev-temp

File: reqs/requirements-dev-contrib.txt (hunk 2 above)
Comment (anthonyburdi) [reply reqs/requirements-dev-contrib.txt]: Consider also moving to requirements-dev-lite.txt or a new requirements-dev-parquet.txt

**Review by anthonyburdi — APPROVED** [review]
LGTM! Thank you for all the work here!

File: ci/azure-pipelines-docs-integration.yml [hunk 13]
```diff
@@ -304,9 +308,13 @@ stages:
           displayName: 'Use Python $(python.version)'
 
         - script: |
-            pip install --constraint constraints-dev.txt ".[dev]" pytest-azurepipelines
-          displayName: 'Install dependencies'
+            cp constraints-dev.txt constraints-dev-temp.txt
+            echo "SQLAlchemy<2.0.0" >> constraints-dev-temp.txt
```
Comment (Shinnnyshinshin) [inline ci/azure-pipelines-docs-integration.yml]: TODO: make a `contraints-sqlalchemy-2.txt` and run them instead.

Comment (Shinnnyshinshin) [reply ci/azure-pipelines-docs-integration.yml]: done

File: tests/datasource/fluent/test_schemas.py [hunk 14]
```diff
@@ -83,11 +83,17 @@ def _sort_lists(schema_as_dict: dict) -> None:
         """
         key: str
         value: Any
+
         for key, value in schema_as_dict.items():
             if key == "required":
                 schema_as_dict[key] = sorted(value)
             elif key == "anyOf":
-                schema_as_dict[key] = sorted(value, key=_sort_any_of)
+                if isinstance(value, list):
```
Comment (Shinnnyshinshin) [inline tests/datasource/fluent/test_schemas.py:91]: @Kilo59 this was giving some errors were schema items were being passed as a list. Some additional checks were made to ensure they were a list of `dicts` and not `str` which can not be sorted. Could you have a look and make any additional appropriate changes?

**Pull request merged**